pkg/cvo/availableupdates.go

```diff
@@ -182,15 +182,12 @@ func calculateAvailableUpdatesStatus(clusterID string, proxyURL *url.URL, tlsCon
 	}
 }
 
-func checkForUpdate(clusterID string, proxyURL *url.URL, tlsConfig *tls.Config, upstream, arch, channel string, currentVersion semver.Version) ([]cincinnati.Update, error) {
+func checkForUpdate(clusterID string, proxyURL *url.URL, tlsConfig *tls.Config, upstream, arch, channel string, currentVersion semver.Version) ([]cincinnati.Update, *cincinnati.Error) {
```

Comment:
We shouldn't be returning types errors, people who care about the type can check that for themselves.

Reply:
returning `*Error` lets the caller avoid a cast attempt and never-gonna-run code for when that cast fails.  You'd still rather me return `error` and cast?

Reply:
Yeah, i don't think the clients need to care if it's this error type, but just `error`, it's fine if some clients care and they can cast imo.

Reply:
Outside of unit tests, this package is the only consumer.  But rerolled to use `error` and a cast with 363a762 -> 8772ab1.